core/domain/param_domain.py

```diff
@@ -57,25 +86,58 @@ class ParamChange(object):
     """Value object for a parameter change."""
 
     def __init__(self, name, generator_id, customization_args):
-
+        """Initialize a ParamChange object with the specified arguments.
+
+        Args:
+            name: unicode. The name of the parameter.
+                generator_id: unicode. The type of generator used to create the
+                parameter, e.g.,"Copier".
+            customization_args: dict. A dict containing the following keys:
+                (value, parse_with_jinja). `value` specifies the value of the
+                parameter, and `parse_with_jinja` indicates whether parsing is
+                to be done with the Jinja template engine. If the parameter is
+                changed to one amongst several values, this dict contains a
+                list (`list_of_values`) of possible values.
+        """
         # TODO(sll): Check that all required args for customization exist in
         # customization_args.
-
         self._name = name
         self._generator_id = generator_id
         self._customization_args = customization_args
 
     @property
     def name(self):
+        """The name of the changing parameter.
+
+        Returns:
+            unicode. The name of the parameter.
+        """
         return self._name
 
     @property
     def generator(self):
+        """The value generator used to define the new value of the
+            changing parameter.
+
+        Returns:
+            generators.Copier. The generator object for the parameter.
```

Comment:
It's not always Copier, right? The type here should be something like 'subclass of BaseClass', where BaseClass is whatever the common base class is.